### PR TITLE
Issue#194 running without moving

### PR DESCRIPTION
**Pull request by RinneT**
Changed behaviour to the following:
Condition:
Merc is running at the beginning of a turn and has enough AP to continue his path.
Player doesn't order merc to continue to move.

In that case the Merc goes to an idle stance as soon as the user ends the turn.

### CODE REVIEW

**Comment by RinneT** [comment]
I think this is actually the cleanest way to implement it.
There's also a StopSoldier function in Soldier_Control.cc, however using that allows you to move your merc during the ai turn.

File: Build/Tactical/TeamTurns.cc [hunk 1]
```diff
@@ -202,6 +202,11 @@ void EndTurn( UINT8 ubNextTeam )
 		FOR_EACH_IN_TEAM(s, gTacticalStatus.ubCurrentTeam)
 		{
 			s->bMoved = TRUE;
+			// Cancel merc movement if continue path was not used
+			if ( CheckForMercContMove(s)) {
```
Comment (derSteFfi) [inline Build/Tactical/TeamTurns.cc]: I think (at least it looks like) we agreed on newlines before '{'.

**Comment by derSteFfi** [comment]
Hey. Glad you found us. Currently testing, but looks fine at first glance.

**Comment by derSteFfi** [comment]
Sooo, this works if the Mercs is selected when you end the turn. Reason is this line in ```CheckForMercContMove```:
```C
	if (s != GetSelectedMan()) return FALSE;
```
Maybe we can extract the checks we need from CheckForMercContMove and do them directly?

File: Build/Tactical/TeamTurns.cc (hunk 1 above)
Comment (lynxlynxlynx) [inline Build/Tactical/TeamTurns.cc]: I prefer the terser variant, but we should just follow whatever is there already or it will get messy.

**Comment by RinneT** [comment]
Thanks a lot for the feedback. That happens when I don't test my changes properly.
Moved the relevant code directly into the function and changed the bracket for style compliance.

**Comment by derSteFfi** [comment]
Much better :) Still, I observed this only working for certain situations. I could not figure out the pattern, but the source showed you're using guiPathingData to obtain the mercs direction. Thats the global copy for painting the path under the mouse cursor etc.
If you use s->usPathingData, which is the copy of the mercs actual path, it works fine (tested here).

**Comment by RinneT** [comment]
Very well noticed, thanks for the hint! :+1:  I was briefly wondering why I'd use a global variable for individual mercs but then didn't think about it anymore when it just worked.
Thinking about that I found out when it fails:
Merc A has two AP left and is moved first. To do his next step he needs 3 AP.
Merc B has one AP left and needs two AP to do his next step. He is moved last.
Now all mercs will be tested against the 2AP from pathing data.

It's more obvious when mercs are having different stances / going stealth.

What I'm wondering:
The function PtsToMoveDirection asks for an INT8.
usPathingData[x] is INT16
guiPathingData[x] is INT32

**Comment by derSteFfi** [comment]
well, you are right (of course :) ). It doesn't seem senseful to have more than 8 bit there. PtsToMoveDirection calls DirectionInc and this checks for the value to be between 0 and 7. I'm trying around with this, but i'd say it is a seperate thing, for now, we should merge the bugfix.

**Pull request merged**